docs/source/integrations/mlflow.md

```diff
@@ -0,0 +1,290 @@
+
+Again, `kedro-mlflow` provides some out-of-the-box artifact tracking capabilities
+that connect your Kedro project with your MLflow deployment.
+
+To that end, introduces a special `MlflowArtifactDataset`
+that can be used to wrap any of your existing Kedro datasets.
+This has the advantage that the preview capabilities of the MLflow UI can be used.
+
+:::{warning}
+This will work for datasets that are outputs of a node,
+and will have no effect for datasets that are free inputs (hence are only loaded).
+:::
+
+For example, if you wrap a `matplotlib.MatplotlibWriter` dataset like this:
+
+```yaml
+dummy_confusion_matrix:
+  type: kedro_mlflow.io.artifacts.MlflowArtifactDataset
+  dataset:
+    type: matplotlib.MatplotlibWriter
+    filepath: data/08_reporting/dummy_confusion_matrix.png
+```
+
+Then the image would be logged as part of the artifacts of the run
+and you would be able to preview it in the MLflow web UI:
+
+```{image} ../meta/images/mlflow-artifact-preview-image.png
+:alt: MLflow image preview thanks to the artifact tracking capabilities of kedro-mlflow
+:width: 80%
+:align: center
+```
+
+Check out {external+kedro-mlflow:doc}`the official kedro-mlflow documentation on versioning Kedro datasets <source/04_experimentation_tracking/03_version_datasets>`
+for more information.
+
+### Model registry in MLflow using `kedro-mlflow`
+
+Finally, if your Kedro pipeline trains a machine learning model, you can track those models in MLflow
+so that you can easily manage and deploy them.
+The `kedro-mlflow` plugin introduces a special artifact, `MlflowModelTrackingDataset`,
+that you can use to load and save your models as MLflow artifacts.
+
+For example, if you have a dataset corresponding to a scikit-learn model,
+you can modify it as follows:
+
+```diff
+ regressor:
+-  type: pickle.PickleDataset
+-  filepath: data/06_models/regressor.pickle
+-  versioned: true
++  type: kedro_mlflow.io.models.MlflowModelTrackingDataset
++  flavor: mlflow.sklearn
+```
+
+The `kedro-mlflow` hook will log the model as part of the run
+in {external+mlflow:doc}`the standard MLflow Model format <models>`.
+
+If, additionally, you want to _register_ it
+(hence store it in the MLflow Model Registry)
+you can add a `registered_model_name` parameter:
+
+```{code-block} yaml
+:emphasize-lines: 4-5
+
+regressor:
+  type: kedro_mlflow.io.models.MlflowModelTrackingDataset
+  flavor: mlflow.sklearn
+  save_args:
+    registered_model_name: spaceflights-regressor
+```
+
+Then you will see it listed as a Registered Model:
+
+```{image} ../meta/images/kedro-mlflow-registered-model.png
+:alt: MLflow Model Registry listing one model registered with kedro-mlflow
+:width: 80%
+:align: center
+```
+
+:::{warning}
+At the moment it's not straightforward to specify the `run_id`
+to use a specific version of a model for inference within a Kedro pipeline,
+see discussion on [this GitHub issue](https://github.com/Galileo-Galilei/kedro-mlflow/issues/549).
```

Comment:
I see people doing  it in production for *a lot* of models, so it is definitely possible ;) I really need to see the code you run to get that error, but essentially you cannot *save* a model in an existing run, because you would alterate it and this is definitely not the purpose of experiment tracking. However, there is absolutely no problem to *load* a model in an existing mlflow run (say the sapecflights regressor you've just trained) to run a kedro pipeline which will log in another mlflow run, or can just be used without logging in mlflow.

Reply:
> essentially you cannot save a model in an existing run, because you would alterate it and this is definitely not the purpose of experiment tracking

That makes a lot of sense. Still chewing on your explanation, will rewrite this part.

Reply:
Addressing this in the upcoming version.